### ISSUE TITLE
Wrong credentials after updating to Windows 11 24H2.

### ISSUE THREAD

**Issue by overclocker-blg** [keepassxreboot/keepassxc#11574]
Hello,
I use KeePassXC for a long time and yesterday I needed a password which I have entered in by database, but when I tried to open my database KeePassXC told me that my credentials are wrong or the database is corrupted. I know that this is the correct password, because I have an earlier copy of the database from 2022 and when I opened this old database with the same password it opened correctly. The problem is that it is an old backup and I need the information stored after 2022 in my main database.

The main database is located on my Synology NAS, in which I have enabled regular snapshots for backup. When I restored the KeepPassXC main database from a couple of months prior I got the same error of invalid credentials or database corruption. I even restored multiple copies dating back to the beginning of 2024, but I could not open my main database.

I want to know if there is a tool with which I can scan my main database and it's older backups to verify if it is corrupted or my credentials are the problem, even that I never changed my password and my old database from 2022 works with that password.

I also want to know if the password is keyboard layout sensitive. For example if English UK and English International layouts are used, some symbols are different, but if I type the correct symbols with a different layout is the password going to be accepted, or I have to use the same layout that I used to create the password?

The only thing that changed during the last time I opened my main database on 19-th of July 2024 was the reinstall of my PC and the update to Windows 24H2 that went 2 days ago.

**Comment by droidmonkey** [comment]
Try typing your password with it visible. 99.9% you are not typing the correct password. Closing this unless there is an obvious fault on our end.

**Comment by overclocker-blg** [comment]
I have tried multiple times even with copy and paste, but still I can't open my main database.

**Comment by droidmonkey** [comment]
You haven't opened it since July, and your backups are not working either, but a really old copy works. That is classic wrong password. You must have changed the database password sometime between 2022 and now and you don't remember doing that.

**Comment by overclocker-blg** [comment]
No I have not changed the password and in that I'm 100% sure. I did not even knew until 2 days ago that I can change a password of a database after creating it.

**Comment by droidmonkey** [comment]
Unfortunately there isn't anything we can do here. To collect a little more information, are you only using a password? No Key file and/or Yubikey? 

Your 2022 copy of the database opens with the credentials.
Your July 2024 and available backups do not open.
You are opening the July 2024 database from your NAS. Is it possible your NAS has actually corrupted your database and files or they did not download correctly... very unlikely but possible.

As mentioned in other threads, the message you receive indicates the beginning header of the database is perfectly fine and we have failed to decrypt / verify the final aspects of it. The probability of just that portion of the database file being corrupt is very very unlikely. The more likely answer is a wrong password.

**Comment by overclocker-blg** [comment]
I'm using a password and a yubikey. The yubikey is inserted, but not visible in KeepassXC until I start the program as administrator. Even started like that it does not accept my password.

2022 is basically the same database with the same yubikey and password made at that time.
July 2024 database is just a copy of the 2022 one with added more data. All other backups from 2024 don't open and I get the message that the credentials are wrong or the database is corrupted on all of them.

The NAS filesystem is scanned once every month for integrity and data is scrubbed. HDD's on the NAS are scanned once a week for problems or bad sectors, even that they are enterprise ones and are used for 13 months. I checked the logs of the NAS and there are no messages that there is a problem with it. Hard drives and filesystem are also OK. I even tested its RAM on the NAS itself, but it also turned OK. 
When I copy a file from the snapshot backups I don't download it with browser, but just copy it from the NAS itself into a working folder to try again the password with KeePassXC so no downloads or moving the database in windows occurs. I copied the 2022 one the same way and it turned out OK.

I know it seems that the most probable answer is wrong password. I haven't slept for 2 days and read many posts here and what I could find on google. That is why I decided to ask if there is a tool with which I can scan the database for corruption, because I'm out of ideas.

**Comment by droidmonkey** [comment]
Are you using the correct slot in the yubikey? Just a wild suggestion

**Comment by overclocker-blg** [comment]
I have both slots configured in YubiKey Manager, but I don't know which one KeePassXC uses.
I'm using Yubikey 5 NFC if that matters.

**Comment by droidmonkey** [comment]
You can try both, recommend doing that, with the same password. No harm in doing that.

**Comment by overclocker-blg** [comment]
Just did that. No luck. Long or short press gives the same result.

Also noticed that on the old database from 2022 in database security there is no yubikey listed. It just shows to add challenge-responce. I don't know if it has to be like that, or has to list that there is already a yubikey associated with this database. Just decided to share if it has any value.

**Comment by droidmonkey** [comment]
if there is no yubikey listed then that means you are not using a yubikey to secure your database.

**Comment by overclocker-blg** [comment]
Thank you for that information. At least there is some progress. So it turns out that my databases were all using only password, since I never went to the Database Security tab to explicitly add the yubikey. I just assumed that on the older versions of KeePass (mainly 2.6.6 when I created the 2022 database) yubikey was detected automatically when trying to open the database and I entered my password and tapped the yubikey every time to let me in and I assumed the yubikey has to be used to secure the database, but as it turns out it was not
